superset/viz.py

```diff
@@ -597,7 +601,13 @@ def get_data(self):
             aggfunc=self.form_data.get('pandas_aggfunc'),
             margins=True,
         )
+
+        # replace line breaks(\n) with <br>
+        for col in df.columns:
+            df[col] = df[col].astype(np.str).str.replace('\n', '<br>')
+
         return df.to_html(
+            escape=False,
```

Comment:
Tested this and checked the docs. Looks like the DataTables javascript renderer disallows xss attacks via this vector. 
https://datatables.net/manual/security#Cross-Site-Scripting

<img width="1422" alt="screenshot 2017-02-06 10 58 33" src="https://cloud.githubusercontent.com/assets/130878/22661962/279e24f8-ec5c-11e6-9eb5-2a54eb947d95.png">